src/js/parsers/section.js

```diff
@@ -148,6 +154,28 @@ class SectionParser {
       return;
     }
 
+    // handle closing the current section and starting a new one if we hit a
+    // new-section-creating element. Skip if we're currently in a list-item
+    // section so that nested lists can be combined
```

Comment:
@kevinansfield is there a test case for this `Skip if we're currently in a list-item section so that nested lists can be combined`? I don't think so, can it be added?

Reply:
@mixonic you're absolutely right, nested grouping wasn't handled at all! I've removed the comment and added a commit with tests of the existing list parsing behaviour to show that it hasn't changed.

I'll push another commit shortly which has a fix for nested lists too but it required some further modification of the section parser. It also has some behaviour changes which _may_ be breaking so I'll comment on those separately.